osxphotos/fileutil.py

```diff
@@ -2,21 +2,233 @@
 
 import datetime
 import enum
+import errno
 import fcntl
+import functools
 import logging
 import os
 import pathlib
 import shutil
 import stat
+import subprocess
 import tempfile
+import types
 import typing as t
 from abc import ABC, abstractmethod
 from tempfile import TemporaryDirectory
 
+from tenacity import (
+    RetryCallState,
+    after_log,
+    before_log,
+    retry,
+    retry_if_exception,
+    stop_after_attempt,
+    wait_fixed,
+)
+
 from .imageconverter import ImageConverter
 from .platform import is_macos
 from .unicode import normalize_fs_path
 
+# logger
+logger = logging.getLogger(__name__)
+
+
+# retry configuration for fileutil operations
+# applicable when --retry and --retry-nas-alias are provided. --retry-wait is optional
```

Comment:
fileutil.py should be independent of the actual command line options used to configure it. Don't reference --retry, etc in fileutil.py, instead explain what the arguments to cfg_fileutil_retry mean. In other words, fileutil.py shouldn't know anything about export.py. It's used elsewhere in the code, not just for export.